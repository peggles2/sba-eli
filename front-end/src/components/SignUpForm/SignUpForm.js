import React from 'react';
import axios from 'axios';
import {Input, Container} from 'semantic-ui-react';
var AmazonCognitoIdentity = require('amazon-cognito-identity-js');

var poolData = {
  UserPoolId : 'us-east-1_B8LlfTU7W', // Your user pool id here
  ClientId : '76jcq337aei27va9q9gtaep0dh' // Your client id here
};
 var userPool = new AmazonCognitoIdentity.CognitoUserPool(poolData);

class SignUpForm extends React.Component {
  state = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: ''
  };

  handleFirstNameChange = (e) => {
    this.setState({firstName: e.target.value});
  }

  handleLastNameChange = (e) => {
    this.setState({lastName: e.target.value});
  }

  handleEmailChange = (e) => {
    this.setState({email: e.target.value});
  }

  handlePasswordChange = (e) => {
    this.setState({password: e.target.value});
  }

  handleConfirmPasswordChange = (e) => {
    this.setState({confirmPassword: e.target.value});
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const url = process.env.REACT_APP_SERVICE_HOST + "/sign_up"

    axios.post(url, {
      first_name: this.state.firstName.trim(),
      last_name: this.state.lastName.trim(),
      email: this.state.email.trim(),
      password: this.state.password.trim(),
      password_confirmation: this.state.confirmPassword.trim(),
    }).then(() => {
      console.log('Success!');
    }).catch((e) => {
      console.log('Failure!', e);
    });
  }

  render() {
    return (
      <Container>
        <form onSubmit={this.handleSubmit.bind(this)}>
          <Input type="text"
                 value={this.state.firstName}
                 placeholder="First Name"
                 onChange={this.handleFirstNameChange.bind(this)}/> <br />
          <Input type="text"
                 value={this.state.lastName}
                 placeholder="Last Name"
                 onChange={this.handleLastNameChange.bind(this)}/> <br />
          <Input type="text"
                 value={this.state.email}
                 placeholder="Email"
                 onChange={this.handleEmailChange.bind(this)}/> <br />
          <Input type="password"
                 value={this.state.password}
                 placeholder="Password"
                 onChange={this.handlePasswordChange.bind(this)}/><br />
          <Input type="password"
                 value={this.state.confirmPassword}
                 placeholder="Confirm Password"
                 onChange={this.handleConfirmPasswordChange.bind(this)}/><br />
          <Input type="submit"/>
        </form>
      </Container>
      
    );
  }
}

export default SignUpForm;