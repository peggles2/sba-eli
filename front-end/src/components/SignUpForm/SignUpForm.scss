@import "../../assets/styles/base";

.noError {
  display: none;
}

#registrationModal {
  .close-icon {
    margin-right: 0;
    padding-right: 0;
  }

  .ui.message {
    padding: .8em 1.5em;
  }

  .horizontalQuestion {
    padding-top: 5px;
  }

  .switchToLogin {
    padding-top: 10px;
  }

  .registrationActions {
    padding-top: 10px;

    .button.submit {
      margin-left: 18px;
    }
  }
}