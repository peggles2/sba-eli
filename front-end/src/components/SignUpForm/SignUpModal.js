import React from 'react';
import axios from 'axios';
import {Modal, Button} from 'semantic-ui-react';
import SignUpForm from './SignUpForm';

class SignUpModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = { open: props.open };
  }

  close = () => {
    this.setState({ open: false });
  }

  open = () => {
    this.setState({ open: true });
  }

  render() {
    return <Modal trigger={<Button onClick={this.open}>Register</Button>}
                  open={this.state.open} 
                  closeOnDimmerClick={false} 
                  closeIcon 
                  onClose={this.close}>
      <Modal.Content>
        <SignUpForm/>
      </Modal.Content>
    </Modal>
  }
}

export default SignUpModal