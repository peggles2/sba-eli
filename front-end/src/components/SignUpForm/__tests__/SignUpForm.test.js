import React from "react";
import { form, List } from "semantic-ui-react";
import SignUpForm from "../SignUpForm";
import { shallow, mount } from "enzyme";

describe("SignUpForm", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<SignUpForm />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render a <form>", () => {
    const wrapper = shallow(<SignUpForm />);

    expect(wrapper.find("Form").length).toEqual(1);
  });
})