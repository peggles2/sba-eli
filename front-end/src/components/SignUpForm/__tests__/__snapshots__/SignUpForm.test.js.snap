// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <Input
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value=""
        />
         
        <br />
        <Input
          onChange={[Function]}
          placeholder="Last Name"
          type="text"
          value=""
        />
         
        <br />
        <Input
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
         
        <br />
        <Input
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <br />
        <Input
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
        />
        <br />
        <Input
          type="submit"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />,
          " ",
          <br />,
          <Input
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
          />,
          " ",
          <br />,
          <Input
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />,
          " ",
          <br />,
          <Input
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />,
          <br />,
          <Input
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />,
          <br />,
          <Input
            type="submit"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "First Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Last Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <Input
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />
           
          <br />
          <Input
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
          />
           
          <br />
          <Input
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
           
          <br />
          <Input
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <br />
          <Input
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
          <br />
          <Input
            type="submit"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
            />,
            " ",
            <br />,
            <Input
              onChange={[Function]}
              placeholder="Last Name"
              type="text"
              value=""
            />,
            " ",
            <br />,
            <Input
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />,
            " ",
            <br />,
            <Input
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <br />,
            <Input
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />,
            <br />,
            <Input
              type="submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "First Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Last Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
